app/src/main/java/com/example/plantsapp/view/PlantsFragment.kt

```diff
@@ -37,13 +37,31 @@ class PlantsFragment : Fragment() {
             plants.observe(viewLifecycleOwner) {
                 plantsAdapter.submitList(it)
             }
+
+            clickedPlant.observe(viewLifecycleOwner) {
+                it.getContentIfNotHandled()?.let {
+                    openFragment(PlantDetailFragment())
+                }
+            }
+
+            isAddPlantClicked.observe(viewLifecycleOwner) {
+                it.getContentIfNotHandled()?.let { isAddBtnClicked ->
+                    if (isAddBtnClicked) {
+                        openFragment(PlantCreationFragment())
+                    }
+                }
+            }
```

Comment:
And instead of boolean you can use just Unit - because you do not need any checks like ISBUTTONCLICKED
Because if event came here - this means that button clicked